### PR TITLE
Sponsors page

### LINKED ISSUE #23: Sponsor Table Component

### PR DESCRIPTION
**Pull request by ElijahMSmith**
Added table for organizing sponsors and a popup for editing or deleting sponsors.

Closes #23 and #24.

### CODE REVIEW

File: src/models/sponsor.ts [hunk 1]
```diff
@@ -13,7 +13,25 @@ export interface Sponsor {
...
 export enum SubscriptionTier {
 	BRONZE,
+	SILVER,
 	GOLD,
 	PLATINUM,
-	SILVER,
+	INVALID,
+}
+
+export function getTierString(tier: SubscriptionTier) {
```
Comment (LightBounded) [inline src/models/sponsor.ts]: We can refactor these functions to be more simple.
`export enum SubscriptionTier {
  BRONZE = "bronze",
  SILVER = "silver",
  GOLD = "gold",
  PLATINUM = "platinum",
  INVALID = "invalid",
}

export function getTierString(tier: SubscriptionTier) {
  return tier.charAt(0).toUpperCase() + tier.slice(1);
}

export function getTierFromString(tier: string) {
  tier = tier.toLowerCase();
  return SubscriptionTier[tier];
}
`

**Review by LightBounded — DISMISSED** [review]
Everything looks good, but we can simplify the way the SubcriptionTier enum is handled

**Review by conroyr41 — CHANGES_REQUESTED** [review]
I believe that the sponsors count should update according to the number of sponsors in the table. But right now it just shows as "0" even though the table is populated with mock data.

image

**Comment by ElijahMSmith** [comment]
> Everything looks good, but we can simplify the way the SubcriptionTier enum is handled

The code you're highlighting/fixing is from an old commit. That translation is not part of the most recent version of sponsor.ts and should be handled more elegantly now.

**Comment by ElijahMSmith** [comment]
> I believe that the sponsors count should update according to the number of sponsors in the table. But right now it just shows as "0" even though the table is populated with mock data.
> 
> image

We can definitely do this, but right now this dummy data only exists in this component. I think we should wait until we have some of the API calls built in and have access to the hackathon data that we are viewing, which we can pass around to each page and to the header cards that display those statistics.

**Review by conroyr41 — APPROVED** [review]
Looks good!

**Pull request merged**